src/django_components/slots.py

```diff
@@ -565,10 +564,7 @@ def on_node(entry: NodeTraverse) -> None:
         _report_slot_errors(slots, slot_fills, component_name)
 
     # 5. Find roots of the slot relationships
-    top_level_slot_ids: List[SlotId] = []
-    for node_id, slot in slots.items():
```

Comment:
Second part of the issue was in this logic that detects top-level slots (AKA slots that are NOT nested in different slots).